pkg/clusteragent/externalmetrics/autoscaler_watcher.go

```diff
@@ -258,7 +261,7 @@ func (w *AutoscalerWatcher) getAutoscalerReferences() (map[string]*externalMetri
 				continue
 			}
 			for _, metric := range wpa.Spec.Metrics {
-				autoscalerReference := wpa.Namespace + kubernetesNamespaceSep + wpa.Name
+				autoscalerReference := autoscalerHPAKindKey + autoscalerReferencesKindSep + wpa.Namespace + kubernetesNamespaceSep + wpa.Name
```

Comment:
typo: `autoscalerHPAKindKey` -> `autoscalerWPAKindKey`

Reply:
good catch

Reply:
@vboulineau  pr updated

Reply:
You need to update unit tests to match (cf. CI failures)